launch/launch/actions/include_launch_description.py

```diff
@@ -122,9 +115,15 @@ def describe_sub_entities(self) -> List[LaunchDescriptionEntity]:
     def visit(self, context: LaunchContext) -> List[LaunchDescriptionEntity]:
         """Override visit to return an Entity rather than an action."""
         launch_description = self.__launch_description_source.get_launch_description(context)
-        context.extend_locals({
-            'current_launch_file_directory': self._get_launch_file_location(),
-        })
+        location = self._get_launch_file_location()
+        if os.path.exists(location):
+            # If the location does not exist, then it's likely set to '<script>' or something.
+            context.extend_locals({
+                'current_launch_file_path': location,
+            })
+            context.extend_locals({
+                'current_launch_file_directory': os.path.dirname(location),
+            })
```

Comment:
Why move this logic here? Also as currently written when it is `<string>` the `current_launch_file_directory` is never set, whereas before it was.

I think it should always be set, just like `__file__` is always set in python even if you're in an interpreter (in which case it uses `<string>` as well).